Lib/gzip.py

```diff
@@ -435,6 +440,13 @@ def readline(self, size=-1):
         self._check_not_closed()
         return self._buffer.readline(size)
 
+    def __del__(self):
+        if self.mode == WRITE and not self.closed:
+            import warnings
+            warnings.warn("unclosed GzipFile",
+                          ResourceWarning, source=self, stacklevel=2)
+
+        return super().__del__()
```

Comment:
I don't think that `__del__()` has a return value.

```suggestion
        super().__del__()
```